pivotal-import/initialize.py

```diff
@@ -159,12 +157,159 @@ def pt_state_mapping_for_workflow(workflow_id):
+    stories), make sure you either (a) run this importer while still in your Shortcut trial
+    period, or (b) reach out to support@shortcut.com to request assistance so you're not
+    billed for extraneous users.
+    """
+    try:
+        with open(users_csv_file, "x") as f:
+            pt_all_users = sorted(extract_pt_users(pt_csv_file))
+            unmapped_pt_users = []
+            sc_user_to_email = fetch_sc_user_to_email()
+            writer = csv.DictWriter(f, ["pt_user_name", "shortcut_user_email"])
+            writer.writeheader()
+            for pt_user in pt_all_users:
+                email = sc_user_to_email.get(pt_user, "")
+                if email == "":
+                    unmapped_pt_users.append(pt_user)
+                writer.writerow({"pt_user_name": pt_user, "shortcut_user_email": email})
+            if unmapped_pt_users:
+                exit_unmapped_pt_users(
+                    users_csv_file, unmapped_pt_users, sc_user_to_email
+                )
+    except FileExistsError:
+        logger.debug("[NOT IMPLEMENTED] populate_users_csv when exists")
+    return 0
+
+
+def identity(x):
+    return x
+
+
+def parse_comment_author(s):
+    """
+    Extract the author from a Pivotal comment.
+
+    Returns None if the comment wasn't parsable.
+    """
+    comment_data = parse_comment(s)
+    if comment_data is not None:
+        return comment_data["author"]
+    else:
+        return None
+
+
+user_cols = {
+    "requested by": identity,
+    "owned by": identity,
+    "reviewer": identity,
+    "comment": parse_comment_author,
+}
+
+
+def extract_pt_users_from_row(row, header):
+    """
+    Extract all Pivotal users from a given CSV row.
+
+    Pivotal's CSV is structured such that only one user is specified
+    per "cell" of the sheet.
+
+    Columns that include a user:
+     - Requested By (value itself)
+     - Owned By (value itself)
+     - Reviewer (value itself)
+     - Comment (authorship is indicated by a suffix)
+    """
+    users_in_row = set()
+    for ix, val in enumerate(row):
+        v = val.strip()
+        if not v:
+            continue
+
+        col = header[ix]
+        if col in user_cols:
+            translator = user_cols[col]
+            user = translator(v)
+            if user is not None:
+                users_in_row.add(user)
+    return users_in_row
+
+
+def extract_pt_users(pt_csv_file):
+    pt_users = set()
+    with open(pt_csv_file) as csvfile:
+        reader = csv.reader(csvfile)
+        header = [col.lower() for col in next(reader)]
+        for row in reader:
+            pt_users.update(extract_pt_users_from_row(row, header))
+    return pt_users
+
+
+def fetch_sc_user_to_email():
+    """
+    Returns a dict mapping the Shortcut user's name to their email address
+    for all members in the Shortcut workspace.
+    """
+    return {
+        member["profile"]["name"]: member["profile"]["email_address"]
```

Comment:
It might be worth including the mention name in this map as well based on the export from Ascenco Health [here](https://app.shortcut.com/internal/epic/262838?vc_group_by=day&ct_workflow=all&cf_workflow=500105268#activity-262870)
<img width="178" alt="Screenshot 2024-04-01 at 17 38 39" src="https://github.com/useshortcut/api-cookbook/assets/100585/6d0de69a-14ab-4393-b2c6-ad10a8211161">